mojo/core/broker_castanets.cc

```diff
@@ -463,19 +572,69 @@ void BrokerCastanets::OnBufferSync2d(uint64_t guid_high,
           << ", width: " << width << ", stride: " << stride
           << ", fence_id: " << fence_id;
 
+  void* uncompressed_data = nullptr;
+
+  switch (compression_mode) {
+    case BrokerCompressionMode::ZLIB: {
+      LOG(ERROR) << "Unsupported compression mode";
+      break;
+    }
+    case BrokerCompressionMode::WEBP: {
+      int width = 0;
+      int height = 0;
+      uint8_t* result = WebPDecodeRGBA(
```

Comment:
It seems to need free..
// The returned pointer should be deleted calling WebPFree().

Reply:
It seems that we can decode directly to shared memory without intermediate buffer.
WEBP_EXTERN uint8_t* WebPDecodeRGBAInto(
    const uint8_t* data, size_t data_size,
    uint8_t* output_buffer, size_t output_buffer_size, int output_stride);

Reply:
Updated the patchset to decode directly to shared memory. Kindly take a look.
WebPFree() is used to delete the encoded result ptr.